pkg/jobs/jobsprofiler/jobsprofiler_test: TestTraceRecordingOnResumerCompletion failed

pkg/jobs/jobsprofiler/jobsprofiler_test.TestTraceRecordingOnResumerCompletion [failed](https://tanzanite.cluster.engflow.com/invocations/default/4c6222b7-5ab5-496c-a458-5f8a2f50e55a?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvam9icy9qb2JzcHJvZmlsZXI6am9ic3Byb2ZpbGVyX3Rlc3Q=) on release-24.1.2-rc @ [4037be1494ff8f3873edfa2510a7fa79545b413e](https://github.com/cockroachdb/cockroach/commits/4037be1494ff8f3873edfa2510a7fa79545b413e):


```
=== RUN   TestTraceRecordingOnResumerCompletion
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTraceRecordingOnResumerCompletion3998100160
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    profiler_test.go:215: condition failed to evaluate within 2m0s: from jobs_verification.go:101: expected job status succeeded, but got running
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTraceRecordingOnResumerCompletion3998100160
--- FAIL: TestTraceRecordingOnResumerCompletion (225.77s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/jobs @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTraceRecordingOnResumerCompletion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39771

--- a/pkg/jobs/jobsprofiler/BUILD.bazel
+++ b/pkg/jobs/jobsprofiler/BUILD.bazel
@@ -24,6 +24,10 @@ go_test(
         "main_test.go",
         "profiler_test.go",
     ],
+    exec_properties = select({
+        "//build/toolchains:is_heavy": {"test.Pool": "large"},
+        "//conditions:default": {"test.Pool": "default"},
+    }),
     deps = [
         ":jobsprofiler",
         "//pkg/base",

